tools/src/file_format/analysis.rs

```diff
@@ -316,6 +316,7 @@ pub enum BindingSlotKind {
     InterfaceName,
     /// Callable.
     Method,
+    MethodImpl,
```

Comment:
Can you elaborate on the rationale for creating `Impl` suffixed variants of these types instead of doing something like [adding an implKind property to the slots](https://matrix.to/#/!vDKxYNxlsZYvjSWGBh:mozilla.org/$BFVpunP0N71qh_ChvkD3qFW-wcwvLj_YORjjg1eN-00?via=mozilla.org&via=matrix.org&via=guichard.eu)?

I'm concerned that we potentially end up needing Impl variants for pretty much everything in this list.  (It does get a bit interesting where for some bindings we're talking about a sub-section of an existing function, which is not a concept searchfox currently has, but I think is something we're likely to want longer term.)  And then match arms get a bit ugly, although that isn't experienced in the JS code here because it's using `endsWith`.

If we stick with this, I think it would be good to have explicit comments that identify the difference between the types.  I often realize I probably need to change how I'm modeling something when I go to write up the comments and I find out it's difficult to explain.  I think this case is probably along those lines, as I think our existing binding slot generators are probably all largely already emitting what would be Impl-suffixed here.  The existence of `formatBindingSlotKind` having first figured out if there are 2 paired slots allows us to compensate for this in the UI.

Of course, this isn't something `implKind` on its own fixes.  If it were an `Option` over an enum, that lets the analysis express that it doesn't care, but we do potentially still have to pair things up.  That's why the other [option I was proposing in chat was to put multiple symbols in the same slot](https://matrix.to/#/!vDKxYNxlsZYvjSWGBh:mozilla.org/$pWbNHvLazoNl_gDzeQfOPF7oBFAaHhi8pJoGviQVxX4?via=mozilla.org&via=matrix.org&via=guichard.eu).  That ends up making the concept of binding slots a bit more complicated to deal with for the simple cases, but avoids the need to take extra-steps to perform the aggregation.  To elaborate, that might look like each binding slot having `syms: [{ sym: "SYM", kind: "binding"}, { sym: "OTHERSYM", kind: "wrapper" }]` instead of a `sym`.

Reply:
Oh, I completely forgot that.
Will fix shortly with optional `implKind`

The `syms` would need some more consideration related to overloads, and I'll leave it to other bug, which possibly need to take other language (Java etc) into account.